test/unit/TicketBroker.js

```diff
@@ -257,6 +269,80 @@ contract("TicketBroker", accounts => {
         })
     })
 
+    describe("requestSignersRevocation", () => {
```

Comment:
What do you think about adding a describe block for `isApprovedSigner()` and moving the test cases involving the state transition from `REVOCATION_IN_PROGRESS` -> `REVOKED` in that section (and moving them out of this section)? I mention that because it feels like that particular state transition isn't a direct result of `requestSignersRevocation()` which is just responsible for the state transition to `REVOCATION_IN_PROGRESS`.

I think we could also add two distinct test cases for when the signerRevocationPeriod elapses:
- When `block.number == signer.revocationBlock`
- When `block.number > signer.revocationBlock`

Reply:
> What do you think about adding a describe block for isApprovedSigner() and moving the test cases involving the state transition

I much prefer to stick with tests that cover the "collaboration" of these two functions. Moving them results in two options that I can see:

1. Simply move all state-transition tests under a different describe block - that's fine, but I don't think we gain much.
2. Split tests up such that under `requestSignersRevocation` we run assertions on `revocationBlock`, and then have similar test cases for `isApprovedSigner` that have the same setup, but then assert in `isApprovedSigner`'s output - this solution seems a bit redundant to me.

So I'm actually inclined to leave the grouping as-is, and perhaps to just rename the describe block to `requestSignersRevocation and isApprovedSigner` since it does indeed encapsulate logic from both functions.

> I think we could also add two distinct test cases for when the signerRevocationPeriod elapses

I agree those are two distinct cases, but the reason I haven't added them so far is because I didn't think this level of granularity mattered much in terms of UX and data integrity. I think it's more important to make sure we cover `block.number > signer.revocationBlock`, so perhaps it's a good idea to change the tests to wait `await fixture.rpc.wait(signerRevocationPeriod + 1)`.

wdyt?

Reply:
If `isApprovedSigner()` is fully tested (i.e. all code paths are hit) then I am ok with not having an additional describe block for the function if it results in a lot of duplication. I'd suggest keeping the `requestSignersRevocation` name for the describe block though as `requestSignersRevocation and isApprovedSigner` seems (at least to me) to imply that these are all the unit test cases for these 2 functions when in fact `isApprovedSigner()` is also used in the describe block for `approveSigners()` which includes some unit test cases for `isApprovedSigner()` as well. Keeping the name seems ok to me.

> I agree those are two distinct cases, but the reason I haven't added them so far is because I didn't think this level of granularity mattered much in terms of UX and data integrity.

I like covering both distinct cases for the sake of completeness i.e. are we absolutely certain that the same result holds true for both distinct cases?